redash/models.py

```diff
@@ -1000,6 +1007,29 @@ def outdated_queries(cls):
 
         return outdated_queries.values()
 
+    @classmethod
+    def delete_stale_resultsets(cls):
+        delete_count = 0
+        queries = Query.query.filter(Query.schedule_keep_results != None).order_by(Query.schedule_keep_results.desc())
+        if queries.first() and queries[0].schedule_keep_results:
+            resultsets = QueryResultSet.query.filter(QueryResultSet.query_rel == queries[0]).order_by(QueryResultSet.result_id)
+            c = resultsets.count()
+            if c > queries[0].schedule_keep_results:
+                n_to_delete = c - queries[0].schedule_keep_results
+                r_ids = [r.result_id for r in resultsets][:n_to_delete]
+                delete_count = QueryResultSet.query.filter(QueryResultSet.result_id.in_(r_ids)).delete(synchronize_session=False)
+                print "one", delete_count
+                QueryResult.query.filter(QueryResult.id.in_(r_ids)).delete(synchronize_session=False)
+        for q in queries[1:]:
```

Comment:
I don't understand why there is another loop here?

Reply:
comments added